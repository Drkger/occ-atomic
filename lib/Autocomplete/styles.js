import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
