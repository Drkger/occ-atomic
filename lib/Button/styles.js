import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';

export default {
    btn: {
        alignItems:'flex-start',
        boxSizing:'border-box',
        display:'inline-block',
        marginBottom:'0',
        textAlign:'center',
        whiteSpace:'nowrap',
        verticalAlign:'middle',
        touchAction:'manipulation',
        cursor:'pointer',
        userSelect:'none',
        padding:'0 30px',
        height:'30px',
        lineHeight:'24px',
        fontFamily:fonts.body,
        fontSize:'14px',
        fontWeight:'600',
        position:'relative',
        transition:'0.3s all',
        maxWidth:'100%',
        borderRadius:'5px',
        textDecoration:'none',
        '&:focus': {
            outline:'none'
        },
        '&:hover': {
            textDecoration:'none'
        }
    },
    cont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        pointerEvents:'none'
    },
    loadCont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        position:'absolute',
        top:'0',
        left:'0',
        pointerEvents:'none'
    },
    // Themes
    main: {
        background:colors.pink,
        color:colors.white,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    secondary: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    white: {
        background:'transparent',
        color:colors.white,
        border:`1px solid ${colors.white}`,
        '&:hover': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:'rgba(255,255,255,0.2)',
            borderColor:colors.white
        },
        '&$disabled': {
            background:`transparent !important`,
            color:`rgba(255,255,255,0.6) !important`,
            border:`1px solid rgba(255,255,255,0.6) !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon(['rgba(255,255,255,0.6)'])) : ''
            }
        }
    },
    flat: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid transparent`,
        '&:hover': {
            background:colors.grey3,
            color:colors.pink,
            borderColor:colors.grey3,
        },
        '&:active': {
            background:colors.grey5,
            color:colors.pink,
            borderColor:colors.grey5,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.pink]))
        },
        '&$loading': {
            background:colors.white,
            borderColor:colors.white
        },
        '&$disabled': {
            background:'transparent !important',
            color:`${colors.grey1} !important`,
            border:`1px solid transparent !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.grey1])) : ''
            },
            '& $loadIcon': {
                background: icons.base(icons.loading.icon([colors.grey1]))
            },
        }
    },
    blue: {
        background:colors.blue,
        color:colors.white,
        border:`1px solid ${colors.blue}`,
        '&:hover': {
            background:colors.blueH,
            color:colors.white,
            borderColor:colors.blueH,
        },
        '&:active': {
            background:colors.blueA,
            color:colors.white,
            borderColor:colors.blueA,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.blueA,
            borderColor:colors.blueA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    loading: {
        pointerEvents:'none',
        '& $cont': {
            visibility:'hidden'
        }
    },
    disabled: {
        background:colors.grey1,
        color:colors.white,
        border:`1px solid ${colors.grey1}`,
        cursor:'default',
        pointerEvents:'none',
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&:hover, &:active, &:focus': {
            background:colors.grey1,
            color:colors.white,
            border:`1px solid ${colors.grey1}`
        }
    },
    // Sizes
    md: {
        fontSize:'16px',
        height:'40px',
        lineHeight:'28px',
        '& i': {
            width:'28px',
            height:'28px'
        }
    },
    lg: {
        fontSize:'18px',
        height:'46px',
        lineHeight:'32px',
        '& i': {
            width:'32px',
            height:'32px'
        }
    },
    // Block
    block: {
        display:'block',
        width:'100%'
    },
    // Icon
    icon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all'
    },
    '@keyframes rotate': {
        from: {transform: 'rotate(0deg)'},
        to: {transform: 'rotate(360deg)'}
    },
    loadIcon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all',
        animation:'rotate 1s infinite linear',
        background: icons.base(icons.loading.icon())
    }
};