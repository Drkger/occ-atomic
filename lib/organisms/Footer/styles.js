import colors from '../../subatomic/colors';
import fonts from '../../subatomic/fonts';
import grid from '../../subatomic/grid';
import icons from '../../subatomic/icons';

export default {
    footer: {
        backgroundColor:colors.black,
        height:'25px',
        margin:'0',
        padding:'0 40px',
        fontSize:'12px',
        color:colors.grey2,
        fontFamily:fonts.title
    },
    footerList: {
        listStyle:'none',
        width:'auto',
        margin:'0',
        padding:'4px 0 0 0'
    },
    listElement: {
        listStyle:'none',
        float:'left',
        margin:'0 50px 0 0',
        padding:'0',
        position:'relative'
    },
    link: {
        color:colors.grey2,
        cursor:'pointer',
        textDecoration:'none',
        '&:hover, &:active, &:focus': {
            color:colors.white,
            textDecoration:'none'
        }
    },
    [`@media (max-width: ${grid.md-1}px)`]: {
        footer: {
            display: 'none !important'
        }
    },
    dropdownMenu: {
        display:'none',
        background:colors.black,
        position:'absolute',
        bottom:'calc(100% + 10px)',
        left:'50%',
        transform:'translateX(-50%)',
        minWidth:'160px',
        padding:'5px 0',
        fontSize:'12px',
        textAlign:'left',
        listStyle:'none',
        borderRadius:'4px',
        boxShadow:'0 6px 12px rgba(0,0,0,.175)',
        zIndex:'2'
    },
    dropdownElement: {
        float:'left',
        listStyle:'none'
    },
    dropdownLink: {
        color:colors.grey2,
        textDecoration:'none',
        padding:'3px 18px',
        lineHeight:'1.42857143',
        display:'block',
        '&:hover': {
            color:colors.white,
            textDecoration:'none'
        }
    }
};