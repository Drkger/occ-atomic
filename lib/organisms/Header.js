import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';
import Waypoint from 'react-waypoint';

import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import grid from '../subatomic/grid';
import icons from '../subatomic/icons';
import shadows from '../subatomic/shadows';

import Container from '../elements/layout/Container';
import Hidden from '../elements/layout/Hidden';
import Visible from '../elements/layout/Visible';
import SightLogo from '../elements/SightLogo';
import UserMenu from '../molecules/UserMenu';
import HeadMenu from '../molecules/HeadMenu';

const styles = {
    navbar: {
        border:'none',
        fontFamily:fonts.body,
        position:'relative',
        zIndex:'9'
    },
    shadow: {
        boxShadow:`${shadows.lvl3} !important`
    },
    master: {
        backgroundColor:colors.blue,
        padding:'0 15px 0 15px',
        minHeight:'60px',
        boxShadow:shadows.lvl3
    },
    content: {
        borderBottom:'1px solid transparent',
        boxShadow:'0px 0px 0px 0px rgba(0, 0, 0, 0.75)'
    },
    title: {
        width:'calc(100% - 150px)',
        fontSize:'16px',
        color:colors.white,
        paddingTop:'17px',
        paddingBottom:'5px',
        float:'left',
        overflow:'hidden',
        textOverflow:'ellipsis',
        whiteSpace:'nowrap'
    },
    user: {
        paddingTop:'9px',
        float:'right !important'
    },
    messages: {
        cursor:'pointer',
        outline:'0',
        display:'inline-flex',
        transform:'translateY(30%)'
    },
    bubble: {
        width:icons.bubble.width,
        height:icons.bubble.height,
        display:icons.bubble.display,
        background:icons.base(icons.bubble.icon())
    },
    separate: {
        marginTop:'60px',
        zIndex:'9',
        position:'relative'
    },
    [`@media (max-width: ${grid.sm-1}px)`]: {
        master: {
            position:'fixed',
            top:'0',
            left:'0',
            width:'100%',
            zIndex:'10',
            boxShadow:'none'
        },
        content: {
            borderTop:'1px solid transparent'
        }
    },
    [`@media (min-width: ${grid.lg}px)`]: {
        master: {
            padding:'0 45px 0 45px'
        }
    }
};

class Header extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {
            shadow: true
        };
        this.removeShadow = this.removeShadow.bind(this);
        this.addShadow = this.addShadow.bind(this);
    }
    
    removeShadow() {
        this.setState({
            shadow: false
        });
    }
    
    addShadow() {
        this.setState({
            shadow: true
        });
    }
    
    render() {
        const { shadow } = this.state;
        const { classes, title, config, recruiterData, validEmail, showR11, navigate, active } = this.props;
        return (
            <div id="mainNav" className={classes.navbar}>
                <div className={`${classes.master} ${shadow ? classes.shadow : ''}`}>
                    <div className={classes.content}>
                        <Hidden xs>
                            <SightLogo />
                        </Hidden>
                        <Visible xs>
                            <div className={classes.title}>{title}</div>
                        </Visible>
                        <div className={classes.user}>
                            <a id="div-r12-header-messages" href={config.ulrChats} className={classes.messages}>
                                <i className={classes.bubble} />
                            </a>
                            <UserMenu {...this.props} />
                        </div>
                        <Hidden xs><HeadMenu active={active} navigate={navigate} /></Hidden>
                    </div>
                </div>
                <Visible xs>
                    <div className={classes.separate}>
                        <Waypoint topOffset={60} onEnter={this.removeShadow} onLeave={this.addShadow}>
                            <div className={!shadow ? classes.shadow : ''}>
                                <HeadMenu active={active} navigate={navigate} />
                            </div>
                        </Waypoint>
                    </div>
                </Visible>
            </div>
        );
    }
}

Header.propTypes = {
    classes: PropTypes.object,
    title: PropTypes.string,
    config: PropTypes.object,
    recruiterData: PropTypes.object,
    validEmail: PropTypes.bool,
    showR11: PropTypes.func,
    navigate: PropTypes.func,
    active: PropTypes.string
};

export default injectSheet(styles)(Header);