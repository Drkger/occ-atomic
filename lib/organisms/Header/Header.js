import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';
import Waypoint from 'react-waypoint';

import Container from '../../elements/layout/Container';
import Hidden from '../../elements/layout/Hidden';
import Visible from '../../elements/layout/Visible';
import SightLogo from '../../elements/SightLogo';

class Header extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {
            shadow: true
        };
        this.removeShadow = this.removeShadow.bind(this);
        this.addShadow = this.addShadow.bind(this);
    }
    
    removeShadow() {
        this.setState({
            shadow: false
        });
    }
    
    addShadow() {
        this.setState({
            shadow: true
        });
    }
    
    render() {
        const { shadow } = this.state;
        const { classes, logo, title, config, recruiterData, validEmail, showR11, navigate, active, nav, menu } = this.props;
        return (
            <div id="mainNav" className={classes.navbar}>
                <div className={`${classes.master} ${shadow ? classes.shadow : ''}`}>
                    <div className={classes.content}>
                        <Hidden xs>
                            {logo}
                        </Hidden>
                        <Visible xs>
                            <div className={classes.title}>{title}</div>
                        </Visible>
                        <div className={classes.user}>
                            <a id="div-r12-header-messages" href={config.ulrChats} className={classes.messages}>
                                <i className={classes.bubble} />
                            </a>
                            {menu}
                        </div>
                        <Hidden xs>{nav}</Hidden>
                    </div>
                </div>
                <Visible xs>
                    <div className={classes.separate}>
                        <Waypoint topOffset={60} onEnter={this.removeShadow} onLeave={this.addShadow}>
                            <div className={!shadow ? classes.shadow : ''}>
                                {nav}
                            </div>
                        </Waypoint>
                    </div>
                </Visible>
            </div>
        );
    }
}

Header.propTypes = {
    classes: PropTypes.object,
    logo: PropTypes.node,
    title: PropTypes.string,
    config: PropTypes.object,
    recruiterData: PropTypes.object,
    validEmail: PropTypes.bool,
    showR11: PropTypes.func,
    navigate: PropTypes.func,
    active: PropTypes.string,
    nav: PropTypes.node
};

export default Header