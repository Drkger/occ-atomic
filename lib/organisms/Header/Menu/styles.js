import colors from '../../../subatomic/colors';
import grid from '../../../subatomic/grid';
import icons from '../../../subatomic/icons';

export default {
    menu: {
        marginTop:'5px',
        display:'inline-block',
        cursor:'pointer !important',
    },
    btnContainer: {
        position:'relative',
        display:'inline-block',
        verticalAlign:'middle'
    },
    btnMenu: {
        cursor:'pointer',
        background:'none',
        border:'none',
        padding:'5px 10px 2px 10px',
        marginLeft:'10px',
        color:colors.white,
        transition:'0.3s all',
        outline:'none !important',
        borderRadius:'3px',
        '&:hover': {
            color:colors.blue,
            background:colors.white
        },
        '&:hover $menuIcon': {
            background:icons.base(icons.menu.icon(["#1476fb"]))
        }
    },
    btnMenuOpen: {
        color:colors.blue,
        background:colors.white,
        '& $menuIcon': {
            background:icons.base(icons.menu.icon(["#1476fb"]))
        }
    },
    userName: {
        paddingRight:'5px',
        margin:'0',
        display:'inline-block',
        overflow:'hidden',
        textOverflow:'ellipsis',
        maxWidth:'65px',
        whiteSpace:'nowrap'
    },
    menuIcon: {
        width:icons.menu.width,
        height:icons.menu.height,
        display:icons.menu.display,
        transition:'0.3s all',
        background:icons.base(icons.menu.icon(["#FFFFFF"]))
    },
    dropdownMenu: {
        position:'absolute',
        top:'100%',
        zIndex:'1000',
        display:'none',
        float:'left',
        minWidth:'160px',
        padding:'5px 0',
        fontSize:'14px',
        textAlign:'left',
        listStyle:'none',
        backgroundColor:colors.white,
        backgroundClip:'padding-box',
        border:'1px solid rgba(0,0,0,.15)',
        boxShadow:'0 6px 12px rgba(0,0,0,.175)',
        margin:'13px 0 0 -187px',
        right:'0'
    },
    dropdownMenuOpen: {
        display:'block'
    },
    listAnchor: {
        textDecoration:'none',
        display:'block',
        padding:'0 10px',
        margin:'5px 0 5px 0',
        clear:'both',
        fontWeight:'400',
        lineHeight:'1.42857143',
        whiteSpace:'nowrap',
        color:'#000',
        cursor:'pointer',
        '&:hover, &:focus': {
            textDecoration:'none',
            color:colors.blue
        }
    },
    signOut: {
        color:`${colors.pink} !important`,
        margin:'10px 0 5px 0'
    },
    separator: {
        height: '1px',
        background:'#ebebeb'
    },
    [`@media (max-width: ${grid.sm-1}px)`]: {
        dropdownMenu: {
            width:'200px'
        }
    }
};