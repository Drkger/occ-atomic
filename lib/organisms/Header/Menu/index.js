import injectSheet from 'react-jss';
import Menu from './Menu';
import styles from './styles';

export default injectSheet(styles)(Menu);