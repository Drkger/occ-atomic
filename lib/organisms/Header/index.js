import injectSheet from 'react-jss';
import Header from './Header';
import styles from './styles';

export default injectSheet(styles)(Header);