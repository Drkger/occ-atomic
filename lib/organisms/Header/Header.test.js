import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';
import styles from './styles';
import HeaderJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Header", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <Header config={{}} classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("HeaderJSS", () => {

    it('matches the snapshot', () => {
        const wrapper = shallow(
            <HeaderJSS config={{}} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});