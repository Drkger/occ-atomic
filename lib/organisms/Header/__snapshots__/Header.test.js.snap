// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    classes={
      Object {
        "@media (max-width: 767px)": "@media (max-width: 767px)",
        "@media (min-width: 1200px)": "@media (min-width: 1200px)",
        "bubble": "bubble",
        "content": "content",
        "master": "master",
        "messages": "messages",
        "navbar": "navbar",
        "separate": "separate",
        "shadow": "shadow",
        "title": "title",
        "user": "user",
      }
    }
    config={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="master shadow"
        >
          <div
            className="content"
          >
            <Hidden
              xs={true}
            />
            <Visible
              xs={true}
            >
              <div
                className="title"
              />
            </Visible>
            <div
              className="user"
            >
              <a
                className="messages"
                href={undefined}
                id="div-r12-header-messages"
              >
                <i
                  className="bubble"
                />
              </a>
            </div>
            <Hidden
              xs={true}
            />
          </div>
        </div>,
        <Visible
          xs={true}
        >
          <div
            className="separate"
          >
            <Waypoint
              bottomOffset="0px"
              fireOnRapidScroll={true}
              horizontal={false}
              onEnter={[Function]}
              onLeave={[Function]}
              onPositionChange={[Function]}
              topOffset={60}
            >
              <div
                className=""
              />
            </Waypoint>
          </div>
        </Visible>,
      ],
      "className": "navbar",
      "id": "mainNav",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="content"
          >
            <Hidden
              xs={true}
            />
            <Visible
              xs={true}
            >
              <div
                className="title"
              />
            </Visible>
            <div
              className="user"
            >
              <a
                className="messages"
                href={undefined}
                id="div-r12-header-messages"
              >
                <i
                  className="bubble"
                />
              </a>
            </div>
            <Hidden
              xs={true}
            />
          </div>,
          "className": "master shadow",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Hidden
                xs={true}
              />,
              <Visible
                xs={true}
              >
                <div
                  className="title"
                />
              </Visible>,
              <div
                className="user"
              >
                <a
                  className="messages"
                  href={undefined}
                  id="div-r12-header-messages"
                >
                  <i
                    className="bubble"
                  />
                </a>
              </div>,
              <Hidden
                xs={true}
              />,
            ],
            "className": "content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "xs": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="title"
                />,
                "xs": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "title",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    className="messages"
                    href={undefined}
                    id="div-r12-header-messages"
                  >
                    <i
                      className="bubble"
                    />
                  </a>,
                  undefined,
                ],
                "className": "user",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <i
                      className="bubble"
                    />,
                    "className": "messages",
                    "href": undefined,
                    "id": "div-r12-header-messages",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "bubble",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                  "type": "a",
                },
                undefined,
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "xs": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="separate"
          >
            <Waypoint
              bottomOffset="0px"
              fireOnRapidScroll={true}
              horizontal={false}
              onEnter={[Function]}
              onLeave={[Function]}
              onPositionChange={[Function]}
              topOffset={60}
            >
              <div
                className=""
              />
            </Waypoint>
          </div>,
          "xs": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Waypoint
              bottomOffset="0px"
              fireOnRapidScroll={true}
              horizontal={false}
              onEnter={[Function]}
              onLeave={[Function]}
              onPositionChange={[Function]}
              topOffset={60}
            >
              <div
                className=""
              />
            </Waypoint>,
            "className": "separate",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bottomOffset": "0px",
              "children": <div
                className=""
              />,
              "fireOnRapidScroll": true,
              "horizontal": false,
              "onEnter": [Function],
              "onLeave": [Function],
              "onPositionChange": [Function],
              "topOffset": 60,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": [Function],
          },
          "type": "div",
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="master shadow"
          >
            <div
              className="content"
            >
              <Hidden
                xs={true}
              />
              <Visible
                xs={true}
              >
                <div
                  className="title"
                />
              </Visible>
              <div
                className="user"
              >
                <a
                  className="messages"
                  href={undefined}
                  id="div-r12-header-messages"
                >
                  <i
                    className="bubble"
                  />
                </a>
              </div>
              <Hidden
                xs={true}
              />
            </div>
          </div>,
          <Visible
            xs={true}
          >
            <div
              className="separate"
            >
              <Waypoint
                bottomOffset="0px"
                fireOnRapidScroll={true}
                horizontal={false}
                onEnter={[Function]}
                onLeave={[Function]}
                onPositionChange={[Function]}
                topOffset={60}
              >
                <div
                  className=""
                />
              </Waypoint>
            </div>
          </Visible>,
        ],
        "className": "navbar",
        "id": "mainNav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="content"
            >
              <Hidden
                xs={true}
              />
              <Visible
                xs={true}
              >
                <div
                  className="title"
                />
              </Visible>
              <div
                className="user"
              >
                <a
                  className="messages"
                  href={undefined}
                  id="div-r12-header-messages"
                >
                  <i
                    className="bubble"
                  />
                </a>
              </div>
              <Hidden
                xs={true}
              />
            </div>,
            "className": "master shadow",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Hidden
                  xs={true}
                />,
                <Visible
                  xs={true}
                >
                  <div
                    className="title"
                  />
                </Visible>,
                <div
                  className="user"
                >
                  <a
                    className="messages"
                    href={undefined}
                    id="div-r12-header-messages"
                  >
                    <i
                      className="bubble"
                    />
                  </a>
                </div>,
                <Hidden
                  xs={true}
                />,
              ],
              "className": "content",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": undefined,
                  "xs": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <div
                    className="title"
                  />,
                  "xs": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "title",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <a
                      className="messages"
                      href={undefined}
                      id="div-r12-header-messages"
                    >
                      <i
                        className="bubble"
                      />
                    </a>,
                    undefined,
                  ],
                  "className": "user",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <i
                        className="bubble"
                      />,
                      "className": "messages",
                      "href": undefined,
                      "id": "div-r12-header-messages",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "bubble",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    "type": "a",
                  },
                  undefined,
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": undefined,
                  "xs": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="separate"
            >
              <Waypoint
                bottomOffset="0px"
                fireOnRapidScroll={true}
                horizontal={false}
                onEnter={[Function]}
                onLeave={[Function]}
                onPositionChange={[Function]}
                topOffset={60}
              >
                <div
                  className=""
                />
              </Waypoint>
            </div>,
            "xs": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Waypoint
                bottomOffset="0px"
                fireOnRapidScroll={true}
                horizontal={false}
                onEnter={[Function]}
                onLeave={[Function]}
                onPositionChange={[Function]}
                topOffset={60}
              >
                <div
                  className=""
                />
              </Waypoint>,
              "className": "separate",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bottomOffset": "0px",
                "children": <div
                  className=""
                />,
                "fireOnRapidScroll": true,
                "horizontal": false,
                "onEnter": [Function],
                "onLeave": [Function],
                "onPositionChange": [Function],
                "topOffset": 60,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              "type": [Function],
            },
            "type": "div",
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`HeaderJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Header)
    config={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "bubble": "Header-bubble-0-1-8",
        "content": "Header-content-0-1-4",
        "master": "Header-master-0-1-3",
        "messages": "Header-messages-0-1-7",
        "navbar": "Header-navbar-0-1-1",
        "separate": "Header-separate-0-1-9",
        "shadow": "Header-shadow-0-1-2",
        "title": "Header-title-0-1-5",
        "user": "Header-user-0-1-6",
      },
      "config": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "bubble": "Header-bubble-0-1-8",
          "content": "Header-content-0-1-4",
          "master": "Header-master-0-1-3",
          "messages": "Header-messages-0-1-7",
          "navbar": "Header-navbar-0-1-1",
          "separate": "Header-separate-0-1-9",
          "shadow": "Header-shadow-0-1-2",
          "title": "Header-title-0-1-5",
          "user": "Header-user-0-1-6",
        },
        "config": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
